### PR TITLE
ColorPalette: Disable `Clear` button if there's no color value.

### LINKED ISSUE #67107: ColorPalette: "Clear" button should be disabled if there's no value.
**Description**

The “Clear” button in the ColorPalette is currently active even when no color value is set. This can be confusing for users, as clicking the button doesn’t do anything in this case. It would be more intuitive if the button were disabled when there’s no color to clear.

https://github.com/user-attachments/assets/18fa5f81-0567-40be-aea1-2efe937c0035

### PR DESCRIPTION
**Pull request by juanfra**
## What?

Fixes https://github.com/WordPress/gutenberg/issues/67107

Disable the "Clear" button if there's no color value for the color palette component.

**EDIT:** I also applied the same logic to the `GradientPicker` and `DuotonePicker` [here](https://github.com/WordPress/gutenberg/pull/67108/commits/50d781e8aae86ade99e7c9b0b80c3ce7daf661ee).

## Why?

The “Clear” button in the ColorPalette is currently active even when no color value is set. This can be confusing for users, as clicking the button doesn’t do anything in this case. It would be more intuitive if the button were disabled when there’s no color to clear.

## How?

By setting it disabled. if there's no value.

## Testing Instructions

**Color picker & gradient picker**

1. Create a page.
2. Add a paragraph.
3. Check the background color, that if there's no background color the "Clear" button is disabled.
4. Check that if there's a background color, the "Clear" button should be active and you could perform the action normally.
5. Repeat with the text color, and confirm the same results.
6. Repeat with the gradient picker.

**Duotone**

1. Create a page, using a theme that has duotone presets.
2. Add an image.
3. Try applying duotones, confirm the clear button is disabled if there's no value.

## Screenshots or screencast 

https://github.com/user-attachments/assets/235ff9ac-b561-4e60-ad8d-ddc25dedac52

### CODE REVIEW

File: packages/components/src/color-palette/index.tsx [hunk 1]
```diff
@@ -249,7 +249,10 @@ function UnforwardedColorPalette(
 	};
 
 	const actions = !! clearable && (
-		<CircularOptionPicker.ButtonAction onClick={ clearColor }>
+		<CircularOptionPicker.ButtonAction
+			onClick={ clearColor }
+			disabled={ ! value }
```
Comment (ntsekouras) [inline packages/components/src/color-palette/index.tsx:255]: Let's also add `accessibleWhenDisabled` prop here.

Comment (juanfra) [reply packages/components/src/color-palette/index.tsx:255]: Thanks for the review. I've added it. I also applied the same logic to the `GradientPicker` and `DuotonePicker`.

**Review by ntsekouras — APPROVED** [review]
LGTM, thank you!

File: packages/components/src/color-palette/index.tsx (hunk 1 above)
Comment (afercia) [reply packages/components/src/color-palette/index.tsx:255]: > Let's also add `accessibleWhenDisabled` prop here.

Thanks for catching that.

**Comment by afercia** [comment]
It's worth reminding why in most cases there is the need to use `accessibleWhenDisabled` when disabling a button.
Under the hood, it uses `aria-disabled` instead of the HTML attribute `disabled`.

When using the keyboard, disabling a button on the fly with `disabled` would make the button no longer focusable _while focus is on it_. As such, there would ba a focus loss. After tabbing again, the tab sequence would start from scratch from the document root (although browsers try to remediate in some way). 

To avoid focus losses and [to make controls discoverable also when disabled](https://www.w3.org/WAI/ARIA/apg/practices/keyboard-interface/#focusabilityofdisabledcontrols), Gutenberg uses the `aria-disabled` pattern + some styling.

**Comment by juanfra** [comment]
Thank you both!

And thanks, @afercia, for the detailed explanation—I’ll definitely keep that in mind moving forward. 🙌

Merging this one now! 🚀

**Pull request merged**